### PR TITLE
#175 add director role privileges

### PR DESCRIPTION
**Pull request by krzyk**
#175 
* added director role privileges
* added tests to cover that
* added todo to add roles to employees and continue implementation

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#279](https://github.com/timelesslounge/timelessis/issues/279) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#279](https://github.com/timelesslounge/timelessis/issues/279) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/7da02978-af55-44df-98b0-62a397c38026); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: timeless/access_control/director_privileges.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     interns.
+    """
+    permitted, user = False, flask.g.get("user")
+    employee_id = kwargs.get("employee_id")
+    if user:
+        if employee_id:
+            if employee_id == user.id and method == Method.READ:
+                permitted = True
+            else:
+                me = Employee.query.get(user.id)
```
Comment (vryazanov) [inline timeless/access_control/director_privileges.py]: @krzyk  `user` is already is an `Employee` instance, no need to query db one more time

**Comment by krzyk** [comment]
@vryazanov updated

**Comment by vryazanov** [comment]
@krzyk please also fix merge conflicts and codeclimate

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 12 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#279](https://github.com/timelesslounge/timelessis/issues/279) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
The job [#279](https://github.com/timelesslounge/timelessis/issues/279) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by vryazanov** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/279#issuecomment-464996077))

@vryazanov This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#279](https://github.com/timelesslounge/timelessis/issues/279) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 8hr
  * @vryazanov[/z](https://www.0crat.com/u/vryazanov) will receive 20 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23279%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@vryazanov According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by vryazanov** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@vryazanov thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable ([here](https://github.com/timelesslounge/timelessis/issues/279#issuecomment-465029033))

@ypshenychka There is no quality review for [#279](https://github.com/timelesslounge/timelessis/issues/279), no performer

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**